pypresence/baseclient.py

```diff
@@ -118,9 +114,18 @@ async def handshake(self):
             raise InvalidPipe
         except asyncio.TimeoutError:
             raise ConnectionTimeout
+    
+    async def handshake(self):
+        ipc_path = get_ipc_path(self.pipe)
+        if not ipc_path:
+            raise DiscordNotFound
+
+        await create_reader_writer(ipc_path)
```

Comment:
Please don't split the handshake function in two functions to keep the pull request as small as possible.
Also this error occoured:
```
Traceback (most recent call last):
  File "/run/media/niklas/Games/pypresence-better/pypresence/test.py", line 14, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/run/media/niklas/Games/pypresence-better/pypresence/test.py", line 8, in main
    await client.start()
  File "/run/media/niklas/Games/pypresence-better/pypresence/pypresence/client.py", line 378, in start
    await self.handshake()
  File "/run/media/niklas/Games/pypresence-better/pypresence/pypresence/baseclient.py", line 123, in handshake
    await create_reader_writer(ipc_path)
          ^^^^^^^^^^^^^^^^^^^^
NameError: name 'create_reader_writer' is not defined
```